inspirehep/modules/workflows/tasks/classifier.py

```diff
@@ -76,33 +76,43 @@ def _classify_paper(obj, eng):
             extract_acronyms=extract_acronyms
         )
 
-        fulltext_used = True
+        try:
+            data = get_value(obj.data, 'titles.title', [])
+            data.extend(get_value(obj.data, 'titles.subtitle', []))
+            data.extend(get_value(obj.data, 'abstracts.value', []))
+            data.extend(get_value(obj.data, 'keywords.value', []))
+            if not data:
+                obj.log.error("No classification done due to missing data.")
+                data_result = None
+            else:
+                data_result = get_keywords_from_text(data, **params)
+        except ClassifierException as e:
+            obj.log.exception(e)
+            data_result = None
+
         with get_document_in_workflow(obj) as tmp_document:
             try:
                 if tmp_document:
                     result = get_keywords_from_local_file(tmp_document, **params)
+                    fulltext_used = True
                 else:
-                    data = get_value(obj.data, 'titles.title', [])
-                    data.extend(get_value(obj.data, 'titles.subtitle', []))
-                    data.extend(get_value(obj.data, 'abstracts.value', []))
-                    data.extend(get_value(obj.data, 'keywords.value', []))
-                    if not data:
-                        obj.log.error("No classification done due to missing data.")
-                        return
-                    result = get_keywords_from_text(data, **params)
+                    result = data_result
                     fulltext_used = False
             except ClassifierException as e:
                 obj.log.exception(e)
                 return
-
-        result['complete_output'] = clean_instances_from_data(
-            result.get("complete_output", {})
-        )
-        result["fulltext_used"] = fulltext_used
-
-        # Check if it is not empty output before adding
-        if any(result.get("complete_output", {}).values()):
-            obj.extra_data['classifier_results'] = result
+        if result:
+            result['complete_output'] = clean_instances_from_data(
+                result.get("complete_output", {})
+            )
```

Comment:
This will only affect `data_result` in case `result == data_result` and L99 is reached. It should be called exactly once for `data_result` in any case.

Reply:
:+1: